### DISCUSSION TITLE
AnalyticalTable: How to Automatically Size Columns to Fit Cell Control?

### DISCUSSION THREAD

**Discussion by albert-li-fioneer** [UI5/webcomponents-react#3844] [Q&A] [answered]
I am working on creating an AnalyticalTable with editable controls of different kinds in the cells. I want the width of the columns to be set so that they fit the control placed in the cell so I set the `scaleWidthMode` property of the table to "Grow". 

According to the description [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable--default-story#properties) this should resize each column to fit the cell contents. However, the columns are not being sized to fit the controls, and parts of them are being cut off:

![image](https://user-images.githubusercontent.com/106703838/206299299-13d731fd-730c-4e4b-8308-544b7e71a04e.png)
Is there something I am missing? Any advice on how I could get the desired behaviour is appreciated.

**Comment by Lukas742** [comment, marked as answer]
Hi @albert-li-fioneer 

the `Grow` and `Smart` mode only work for cells with plain text as content. I see that our documentation is currently not clear enough about this, so I created a [PR](https://github.com/SAP/ui5-webcomponents-react/pull/3849) to describe it better. Cells with content other than text are ignored, that's why in your case only the header text was taken into account and therefore "PCD" is, for example, much more narrow than "EXPIRY_DATE". 
When you're using components as content it's in most cases recommended to set a `width` or maybe even a `minWidth`.

**Reply by albert-li-fioneer** [reply]
Thanks for the response! Is there a way of programmatically obtaining the width of the component in the cell so I can set the column width based on it?

**Reply by Lukas742** [reply]
I wouldn't recommend that. For measuring (e.g. with `offsetWidth`) you need to mount your elements first and then adjust the value inside the table. So you will either have to mount them somewhere else (hidden) and only after they mounted and you received the width, mount the table including the custom components, or measure them while inside the table and update the width in runtime, which needs rerendering for each component.

It get's even worse when you want to dynamically change the content, then you will probably need to register a observer and update the column width on every resize change. 

Both approaches will have a negative impact on performance and maybe even UX.
